libres/lib/enkf/local_obsdata.cpp

```diff
@@ -84,71 +71,79 @@ const char *local_obsdata_get_name(const local_obsdata_type *data) {
 }
 
 int local_obsdata_get_size(const local_obsdata_type *data) {
-    return vector_get_size(data->nodes_list);
+    return data->nodes.size();
 }
 
 /*
-  The @data instance will assume ownership of the node; i.e. calling
-  scope should NOT call local_obsdata_node_free().
+  The @data instance will insert a copy
 */
 
 bool local_obsdata_add_node(local_obsdata_type *data,
-                            local_obsdata_node_type *node) {
-    const char *key = local_obsdata_node_get_key(node);
+                            const LocalObsDataNode *node) {
+    const char *key = node->name().c_str();
     if (local_obsdata_has_node(data, key))
         return false;
     else {
-        vector_append_owned_ref(data->nodes_list, node,
-                                local_obsdata_node_free__);
-        hash_insert_ref(data->nodes_map, key, node);
+        data->node_index.emplace(key, data->nodes.size());
+        data->nodes.push_back(*node);
         return true;
     }
 }
 
 void local_obsdata_del_node(local_obsdata_type *data, const char *key) {
```

Comment:
Is this used? Seems to me it is never used in C++, it is exposed in `python`, but cant really see that it is used there?

Reply:
> Is this used? Seems to me it is never used in C++, it is exposed in python, but cant really see that it is used there?

It is a feature requested by Oddvar and in my opinion it does make sense. While booting up the the global all active configurations are assembled. This can be used to conveniently create a configuration with "All observations *except* obs1 and obs2".

Reply:
Do you remember when it was requested? As far as I can tell it is not being used at the moment, that does not necessarily mean we must delete it, but might run it by Oddvar if the need is no longer there.

Reply:
It was requested when we worked on this ~1 year ago.